### PR TITLE
Admins can now launch tactical nukes at the station

### PR DESCRIPTION
**Pull request by Kmc2000**
### Intent of your Pull Request

The real question is why DON'T we need this.

I cannot make the admin verb modular. The rest is modular.

### Intent of your Pull Request

https://www.youtube.com/watch?v=rN760628dm8

Defer to this video for information:

https://drive.google.com/file/d/1JJI5uRRw1wIB2dtHyx7WopzPfcZtRwmX/view?usp=sharing

This allows for a more atmospheric nuking of the station, the "Immolator" class missile will make all those problems disappear.

Points of note:
. Any unprotected mobs will be vaporized on the initial impact
. You can survive this! a hint on what to do is given in the announcement that accompanies the blast
. Fallout isn't like the weird, traditional radstorm. Instead it just deals rad damage over-time. You can still go out in it but expect to contract cancer.
. Sounds made by myself through various resources on YouTube
. This can only be triggered by admins due to its destructive nature.
. This does not replace the traditional nuke, it's just a bit of fun for events and the like

https://i.gyazo.com/bc7e8808c00423d6b0743466162e7843.mp4

#### Changelog

:cl: Kmc2000  
rscadd: Admins can now launch realistic tactical nukes at the station. (Admins, use the "Tactical Nuke" verb in the fun tab)
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
This is america

**Comment by Keekenox** [comment]
You are an absolute madman KMC.

**Comment by fluffe9911** [comment]
TACTICAL NUKE INBOUND

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -186,7 +187,7 @@ GLOBAL_LIST_INIT(admin_verbs_hideable, list(
 	/client/proc/admin_ghost,
 	/client/proc/toggle_view_range,
 	/client/proc/cmd_admin_subtle_message,
-	/client/proc/cmd_admin_headset_message,	
+	/client/proc/cmd_admin_headset_message,
```
Comment (ThatLing) [inline code/modules/admin/admin_verbs.dm:190]: Put this space back right now, young man.

File: yogstation/code/datums/weather/nuclear_impact.dm [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+/datum/weather/nuclear/telegraph()
+	. = ..()
+	priority_announce("WARNING: TACTICAL NUCLEAR DETONATION DETECTED IN SS13 AIRSPACE. ALL PERSONNEL ARE ADVISED TO SEEK SHELTER. OFFICIAL INSTRUCTIONS WILL FOLLOW.", "Nanotrasen Emergency Alert System")
+	sleep(60)
+	priority_announce("Official advice from Nanotrasen citizen's office: Put as much metal between you and the outside as possible. All space-facing zones are likely to be affected by the initial heatwave.", "Nanotrasen Emergency Alert System")
+
+/datum/weather/nuclear/start()
+	. = ..()
+	for(var/mob/living/L in GLOB.player_list)
```
Comment (ThatLing) [inline yogstation/code/datums/weather/nuclear_impact.dm]: Use ``mob_living_list`` if you need all alive mobs.

File: yogstation/code/modules/admin/verbs/fun.dm [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+		return
+	if(check_rights(R_ADMIN,1))
+		var/our_badmin = usr
+		var/areyouDrunk = input(our_badmin, "Please confirm action, this is highly destructive!", "Centcom Nuclear Failsafe") in list("YES","NO")
+		switch(areyouDrunk)
+			if("YES")
+				log_admin("[key_name(usr)] launched a nuke at the station!", 1)
+				message_admins("<span class='adminnotice'>[key_name_admin(usr)] launched a nuke at the station!</span>")
+				SSweather.run_weather("nuclear detonation",2)
+				return 1
```
Comment (ThatLing) [inline yogstation/code/modules/admin/verbs/fun.dm]: return TRUE

File: yogstation/code/modules/admin/verbs/fun.dm [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+			if("NO")
+				log_admin("[key_name(usr)] Resisted the temptation to press the nuclear launch button.", 1)
+				return 0
```
Comment (ThatLing) [inline yogstation/code/modules/admin/verbs/fun.dm]: return FALSE

**Review by ThatLing — DISMISSED** [review]

File: yogstation/code/datums/weather/nuclear_impact.dm [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	if(isliving(M))
+		if(ishuman(M))
+			var/mob/living/carbon/human/L = M
+			var/resist = L.getarmor(null, "energy")
+			if(prob(max(0,100-resist)))
+				if(/obj/item/clothing/suit/fire || /obj/item/clothing/suit/space in L.wear_suit)
+					L.adjustToxLoss(3)
+					L.IgniteMob()
+				else
+					to_chat(L, "In an instant flash of heat, you are completely vaporized as your skin melts away.")
```
Comment (Amelia0010) [inline yogstation/code/datums/weather/nuclear_impact.dm:55]: they should also be told this if they aren't human

**Review by Amelia0010 — APPROVED** [review]

**Comment by ThatLing** [comment]
Not getting merged until the space in ``code/modules/admin/admin_verbs.dm`` is put back.

File: yogstation/code/datums/weather/nuclear_impact.dm [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+/datum/weather/nuclear/telegraph()
+	. = ..()
+	priority_announce("WARNING: TACTICAL NUCLEAR DETONATION DETECTED IN SS13 AIRSPACE. ALL PERSONNEL ARE ADVISED TO SEEK SHELTER. OFFICIAL INSTRUCTIONS WILL FOLLOW.", "Nanotrasen Emergency Alert System")
+	sleep(60)
+	priority_announce("Official advice from Nanotrasen citizen's office: Put as much metal between you and the outside as possible. All space-facing zones are likely to be affected by the initial heatwave.", "Nanotrasen Emergency Alert System")
+
+/datum/weather/nuclear/start()
+	. = ..()
+	for(var/mob/living/L in GLOB.mob_living_list)
+		shake_camera(L, 7, 1)
```
Comment (ThatLing) [inline yogstation/code/datums/weather/nuclear_impact.dm:73]:
```
for(var/T in GLOB.mob_living_list)
    var/mob/living/L = T
    shake_camera(L, 7, 1)
```

File: yogstation/code/datums/weather/nuclear_impact.dm [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+			else
+				L.IgniteMob()
+		else
+			var/mob/living/themob = M
+			to_chat(M, "In an instant flash of heat, you are completely vaporized as your skin melts away.")
+			themob.dust()
+
+/datum/weather/nuclear/telegraph()
+	. = ..()
+	priority_announce("WARNING: TACTICAL NUCLEAR DETONATION DETECTED IN SS13 AIRSPACE. ALL PERSONNEL ARE ADVISED TO SEEK SHELTER. OFFICIAL INSTRUCTIONS WILL FOLLOW.", "Nanotrasen Emergency Alert System")
```
Comment (ThatLing) [inline yogstation/code/datums/weather/nuclear_impact.dm:66]: muh immersion

File: yogstation/code/datums/weather/nuclear_impact.dm [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+			if(END_STAGE)
+				N.color = null
+				N.icon_state = ""
+				N.icon = 'icons/turf/areas.dmi'
+				N.layer = AREA_LAYER //Just default back to normal area stuff since I assume setting a var is faster than initial
+				N.set_opacity(FALSE)
+
+/datum/weather/nuclear/weather_act(atom/movable/M)
+	if(isliving(M))
+		if(ishuman(M))
```
Comment (ThatLing) [inline yogstation/code/datums/weather/nuclear_impact.dm:47]: Why the ``isliving(M)`` check here?

File: yogstation/code/modules/admin/verbs/fun.dm [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/client/proc/FireNuke()
+	set name = "Tactical Nuke"
+	set category = "Admin"
```
Comment (ThatLing) [inline yogstation/code/modules/admin/verbs/fun.dm:3]: Move this to fun

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: yogstation/code/datums/weather/nuclear_impact.dm (hunk 8 above)
Comment (AsV9) [reply yogstation/code/datums/weather/nuclear_impact.dm:47]: i think we need an `ismob(M)` here too to be honest

**Comment by ThatLing** [comment]
Stale, feel free to re-open or make a new PR when you are working on this again.

**Pull request closed without merging**